src/game.cpp

```diff
@@ -9913,6 +9984,16 @@ static cata::optional<tripoint> point_selection_menu( const std::vector<tripoint
 
 void game::vertical_move( int movez, bool force )
 {
+    if( u.has_effect( effect_riding ) && u.mounted_creature ) {
+        auto mons = u.mounted_creature.get();
+        if( mons->has_flag( MF_RIDEABLE_MECH ) ) {
+            if( !mons->check_mech_powered() ) {
+                add_msg( m_bad, _( "Your %s refuses to move as its batteries have been drained." ),
+                         mons->get_name() );
+                return;
+            }
+        }
+    }
```

Comment:
This condition is checked several times throughout the code. Maybe create a separate bool check for that?

Reply:
Sorry, can't see which condition that was anymore?
If it was has_effect( effect_riding ) && mounted_creature, then I've added that separate function.